diff --git a/arch/arm/rules.mk b/arch/arm/rules.mk
index bfef6825c..f1ad338f7 100644
--- a/arch/arm/rules.mk
+++ b/arch/arm/rules.mk
@@ -20,6 +20,22 @@ DEFINES += \
 	ARM_WITH_CACHE=1 \
 	ARM_WITH_L2=1
 CFLAGS += -mcpu=$(ARM_CPU)
+HANDLED_CORE := true
+endif
+ifeq ($(ARM_CPU),cortex-a53)
+DEFINES += \
+	ARM_WITH_CP15=1 \
+	ARM_WITH_MMU=1 \
+	ARM_ISA_ARMv7=1 \
+	ARM_WITH_VFP=1 \
+	ARM_WITH_NEON=1 \
+	ARM_WITH_THUMB=1 \
+	ARM_WITH_THUMB2=1 \
+	ARM_WITH_CACHE=1 \
+	ARM_WITH_L2=1 \
+	ARM_CPU_CORTEX_A8=1
+CFLAGS += -mcpu=$(ARM_CPU) -march=armv7-a+sec+fp
+HANDLED_CORE := true
 #CFLAGS += -mcpu=arm1136jf-s # compiler doesn't understand cortex yet
 HANDLED_CORE := true
 #CFLAGS += -mfpu=vfp -mfloat-abi=softfp
diff --git a/platform/msm8916/rules.mk b/platform/msm8916/rules.mk
index b0a207001..de6076c46 100644
--- a/platform/msm8916/rules.mk
+++ b/platform/msm8916/rules.mk
@@ -2,7 +2,7 @@ LOCAL_DIR := $(GET_LOCAL_DIR)
 
 ARCH    := arm
 #Compiling this as cortex-a8 until the compiler supports krait
-ARM_CPU := cortex-a8
+ARM_CPU := cortex-a53
 CPU     := generic
 
 DEFINES += ARM_CPU_CORE_A7
diff --git a/target/msm8916/init.c b/target/msm8916/init.c
index 446cfe4e9..344856b1d 100644
--- a/target/msm8916/init.c
+++ b/target/msm8916/init.c
@@ -51,7 +51,6 @@
 #include <partition_parser.h>
 #include <stdlib.h>
 #include <smem.h>
-#include <lk2nd-device.h>
 #include <qtimer.h>
 #include <spmi.h>
 #include <sdhci_msm.h>
