diff --git a/pexpert/conf/files.arm b/pexpert/conf/files.arm
index 6918d22..c8ecbd0 100644
--- a/pexpert/conf/files.arm
+++ b/pexpert/conf/files.arm
@@ -18,4 +18,6 @@ pexpert/arm/pe_s5l8720x.c		standard
 pexpert/arm/pe_s5l8900x.c		standard
 pexpert/arm/pe_s5l8930x.c		standard
 pexpert/arm/pe_apq8060.c		standard
+pexpert/arm/pe_msm8916.c		standard
+pexpert/arm/pe_generic.c		standard
 pexpert/arm/pe_imx53.c			standard
diff --git a/pexpert/conf/files.arm64 b/pexpert/conf/files.arm64
index 3e70b9a..2a689ff 100644
--- a/pexpert/conf/files.arm64
+++ b/pexpert/conf/files.arm64
@@ -6,3 +6,4 @@ pexpert/arm/common/pe_bootargs.c	standard
 pexpert/arm/common/pe_identify_machine.c	standard
 pexpert/arm/common/pe_socsupport.c	standard
 pexpert/arm/common/pe_armsupport.c	standard
+pexpert/arm/pe_generic_arm64.c		standard
